src/Fmt_ast.ml

```diff
@@ -3668,10 +3671,16 @@ and fmt_structure_item c ~last:last_item ?ext {ctx; ast= si} =
       let fmt_decl ~first ~last decl =
         let pre =
           if first then
-            if Poly.(rec_flag = Recursive) then "type " else "type nonrec "
+            let pre =
+              if Poly.(rec_flag = Recursive) then "type" else "type nonrec"
+            in
+            if Option.is_some ext then pre else pre ^ " "
```

Comment:
Minor comment:
This line is a bit inelegant. The `?pre` argument of `fmt_type_declaration` never needs to be `None` and is always terminated by a space character. (Except for `with` constraints but I get no regression) What do you think of adding the space in `fmt_type_declaration`, in every cases?

Reply:
Done.